tenseal/tensealcontext.cpp

```diff
@@ -2,30 +2,27 @@
 
 #include <seal/seal.h>
 
-#include <vector>
-
 #include "utils.h"
 
+namespace tenseal {
+
 using namespace seal;
 using namespace std;
```

Comment:
Q: What's the advantage of putting this here?

Reply:
In the context of cpps, not much.

But it's usually for the best to avoid polluting the global namespace with "using" statements, or at least limit them to the scope of another namespace.
Another reason for using the in the cpp could be for preventing including headers after the using statement. It's usually recommended to add at least a anonymous namespace before adding "using".


In ".h" is not recommended at all, it must be scoped by a namespace, because you change the scope/API for all your integrators.

Reply:
I know for header files, but thought that it doesn't affect for cpps

Reply:
The only issue with cpp files is adding an "using" statement before an "include" statement.
But in general, it can lead to annoying bugs anyway,